starter-core/src/main/java/io/micronaut/starter/feature/oraclecloud/OracleCloudVault.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2017-2021 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.oraclecloud;
+
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.dependencies.Dependency;
+import io.micronaut.starter.feature.distributedconfig.DistributedConfigFeature;
+import jakarta.inject.Singleton;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Singleton
+public class OracleCloudVault implements DistributedConfigFeature {
+
+    @Override
+    public String getName() {
+        return "oracle-cloud-vault";
+    }
+
+    @Override
+    public String getTitle() {
+        return "Oracle Cloud Vault Distributed Configuration";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Adds support for Distributed Configuration with Oracle Cloud Vault";
+    }
+
+    @Nullable
+    @Override
+    public String getMicronautDocumentation() {
+        return "https://micronaut-projects.github.io/micronaut-oracle-cloud/latest/guide/#vault";
+    }
+
+    @Nullable
+    @Override
+    public String getThirdPartyDocumentation() {
+        return "https://docs.oracle.com/en-us/iaas/Content/KeyManagement/home.htm";
+    }
+
+    @Override
+    public void apply(GeneratorContext generatorContext) {
+        generatorContext.addDependency(Dependency.builder()
+                .groupId("io.micronaut.oraclecloud")
+                .artifactId("micronaut-oraclecloud-vault")
+                .compile());
+        generatorContext.getConfiguration().put("oci.config.profile", "DEFAULT");
+        generatorContext.getBootstrapConfiguration().put("micronaut.application.name", generatorContext.getProject().getPropertyName());
```

Comment:
This isn't necessary. I see there are 2 other features that do this but its already handled by the AppName feature